src/jit/emitxarch.cpp

```diff
@@ -5550,11 +5550,27 @@ static bool isSSEExtract(instruction ins)
     }
 }
 
+static bool isSseShuffle(instruction ins)
+{
+    switch (ins)
+    {
+        case INS_pshufb:
+        case INS_pshufd:
+        case INS_pshufhw:
+        case INS_pshuflw:
+        case INS_shufpd:
+        case INS_shufps:
+            return true;
+        default:
+            return false;
+    }
+}
+
 void emitter::emitIns_SIMD_R_R_I(instruction ins, emitAttr attr, regNumber reg, regNumber reg1, int ival)
 {
     // TODO-XARCH refactoring emitIns_R_R_I to handle SSE2/AVX2 shift as well as emitIns_R_I
     bool isShift = isSseShift(ins);
-    if (isSSEExtract(ins) || (UseVEXEncoding() && !isShift))
+    if (isSSEExtract(ins) || isSseShuffle(ins) || (UseVEXEncoding() && !isShift))
```

Comment:
This seems wrong.

`pshufb` is an `SSSE3` instruction and is encoded as `xmm1, xmm2/m128` (`R_R`) or `xmm1, xmm2, xmm3/m128`  (`R_R_R`)

`shufpd` and `shufps` are encoded as `xmm1, xmm2/m128, imm8` (R_R_I) or `xmm1, xmm2, xmm3/m128, imm8` (R_R_R_I)

This is really just checking for `pshufd`, `pshufhw`, or `pshuflw`

Reply:
Good catch - will fix.

Reply:
This was probably not causing a problem because the other instructions are going down seperate `emitIns` code paths (`emitIns_SIMD_R_R_R` and `emitIns_SIMD_R_R_R_I`, respectively)

I would still like to see this fixed up to help increase the code clarity.

Reply:
Fixed in latest commit